[Bug]: request funktioniert nicht mehr

### I'm sure that

- [X] This issue is still present in the **current beta version** of this adapter
- [X] There is no other (open) issue with the same topic (use the search!)
- [X] This issue is not described in the adapter documentation / FAQ (read the docs!)

### Script type

Blockly

### The problem

Seit der Version 8.7.2 ist kein request mehr möglich.

 Es gibt nur eine error Ausgabe.

### iobroker.current.log (in debug mode!)

<html><body>
<!--StartFragment-->
javascript.0 | 143315 | 2024-07-26 10:59:11.821 | error | script.js.Sonnenschutz.Rollo1: TypeError: require(...) is not a function     at Object.<anonymous> (script.js.Sonnenschutz.Rollo1:19:23)     at Object.callback (/opt/iobroker/node_modules/iobroker.javascript/lib/sandbox.js:1439:38)     at Object.stateChange (/opt/iobroker/node_modules/iobroker.javascript/main.js:645:29)     at Immediate._onImmediate (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.ts:11053:62)     at processImmediate (node:internal/timers:476:21)
-- | -- | -- | -- | --
javascript.0 | 143315 | 2024-07-26 10:59:11.820 | error | at processImmediate (node:internal/timers:476:21)
javascript.0 | 143315 | 2024-07-26 10:59:11.819 | error | at Immediate._onImmediate (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.ts:11053:62)
javascript.0 | 143315 | 2024-07-26 10:59:11.819 | error | at Object.stateChange (/opt/iobroker/node_modules/iobroker.javascript/main.js:645:29)
javascript.0 | 143315 | 2024-07-26 10:59:11.819 | error | at Object.callback (/opt/iobroker/node_modules/iobroker.javascript/lib/sandbox.js:1439:38)
javascript.0 | 143315 | 2024-07-26 10:59:11.818 | error | at Object.<anonymous> (script.js.Sonnenschutz.Rollo1:18:5)
javascript.0 | 143315 | 2024-07-26 10:59:11.818 | error | at require (/opt/iobroker/node_modules/iobroker.javascript/lib/sandbox.js:548:32)
javascript.0 | 143315 | 2024-07-26 10:59:11.817 | error | at require (node:internal/modules/helpers:177:18)
javascript.0 | 143315 | 2024-07-26 10:59:11.817 | error | at Module.require (node:internal/modules/cjs/loader:1231:19)
javascript.0 | 143315 | 2024-07-26 10:59:11.816 | error | script.js.Sonnenschutz.Rollo1: Error: Cannot find module '@iobroker-javascript.0/request'
javascript.0 | 143315 | 2024-07-26 10:59:11.809 | error | script.js.Sonnenschutz.Rollo1: TypeError: require(...) is not a function     at Object.<anonymous> (script.js.Sonnenschutz.Rollo1:19:23)     at Object.callback (/opt/iobroker/node_modules/iobroker.javascript/lib/sandbox.js:1439:38)     at Object.stateChange (/opt/iobroker/node_modules/iobroker.javascript/main.js:645:29)     at Immediate._onImmediate (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.ts:11053:62)     at processImmediate (node:internal/timers:476:21)
javascript.0 | 143315 | 2024-07-26 10:59:11.807 | error | at processImmediate (node:internal/timers:476:21)
javascript.0 | 143315 | 2024-07-26 10:59:11.807 | error | at Immediate._onImmediate (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.ts:11053:62)
javascript.0 | 143315 | 2024-07-26 10:59:11.806 | error | at Object.stateChange (/opt/iobroker/node_modules/iobroker.javascript/main.js:645:29)
javascript.0 | 143315 | 2024-07-26 10:59:11.806 | error | at Object.callback (/opt/iobroker/node_modules/iobroker.javascript/lib/sandbox.js:1439:38)
javascript.0 | 143315 | 2024-07-26 10:59:11.806 | error | at Object.<anonymous> (script.js.Sonnenschutz.Rollo1:18:5)
javascript.0 | 143315 | 2024-07-26 10:59:11.805 | error | at require (/opt/iobroker/node_modules/iobroker.javascript/lib/sandbox.js:548:32)
javascript.0 | 143315 | 2024-07-26 10:59:11.805 | error | at require (node:internal/modules/helpers:177:18)
javascript.0 | 143315 | 2024-07-26 10:59:11.805 | error | at Module.require (node:internal/modules/cjs/loader:1231:19)
javascript.0 | 143315 | 2024-07-26 10:59:11.804 | error | script.js.Sonnenschutz.Rollo1: Error: Cannot find module '@iobroker-javascript.0/request'
javascript.0 | 143315 | 2024-07-26 10:59:11.798 | warn | script.js.Sonnenschutz.Rollo1: request package is deprecated - please use httpGet (or a stable lib like axios) instead!
javascript.0 | 143315 | 2024-07-26 10:59:00.048 | info | script.js.TOOLS.Mondstate: Mond: Elevation: 8.4°
javascript.0 | 143315 | 2024-07-26 10:59:00.048 | info | script.js.TOOLS.Mondstate: Mond: Azimut: 265.1°
javascript.0 | 143315 | 2024-07-26 10:59:00.047 | info | script.js.TOOLS.Mondstate: Sonne: Elevation: 47°
javascript.0 | 143315 | 2024-07-26 10:59:00.046 | info | script.js.TOOLS.Mondstate: Sonne: Azimut: 124.9°

<!--EndFragment-->
</body>
</html>

### Version of nodejs

v18.20.4

### Version of ioBroker js-controller

v6.0.9

### Version of adapter

v8.7.2

--- a/README.md
+++ b/README.md
@@ -26,6 +26,10 @@ Executes Javascript, Typescript Scripts.
 <!--
 	### **WORK IN PROGRESS**
 -->
+
+### **WORK IN PROGRESS**
+* (foxriver76) added fallback to require the adapters `request` module if no own installed
+
 ### 8.7.3 (2024-07-25)
 * (@klein0r) Fixed script deletion (icon was missing)
 * (@klein0r) Fixed some translations and Blockly definitions

--- a/lib/sandbox.js
+++ b/lib/sandbox.js
@@ -534,10 +534,15 @@ function sandBox(script, name, verbose, debug, context) {
                 md = md.replace(/^node:/, '');
             }
 
-            if (typeof md === 'string') {
-                if (md === 'request' && !sandbox.__engine.__deprecatedWarnings.includes(md)) {
+            if (md === 'request') {
+                if (!sandbox.__engine.__deprecatedWarnings.includes(md)) {
                     sandbox.log(`request package is deprecated - please use httpGet (or a stable lib like axios) instead!`, 'warn');
                     sandbox.__engine.__deprecatedWarnings.push(md);
+                }
+
+                if (!mods[md]) {
+                    // for backward compatibility we require the adapters own request module as back in the days this module did not need to be specified in the additional npm modules
+                    mods[md] = require(md);
                 }
             }
 

